z3c/dependencychecker/db.py

```diff
@@ -43,3 +48,186 @@ def _filter_duplicates(self, imports):
         all_imports = set([dotted_name for dotted_name in imports])
         filtered = all_imports - self._requirements
         return filtered
+
+    def add_imports(self, imports):
+        for single_import in imports:
+            self.imports_used.append(single_import)
+
+    def get_missing_imports(self):
+        filters = (
+            self._filter_known_packages,
+            self._filter_testing_imports,
+            self._filter_own_package,
+            self._filter_requirements,
+            self._filter_python_standard_library,
+        )
+        filtered = self._process_pipeline(self.imports_used, filters)
+        unique_imports = self._get_unique_imports(imports_list=filtered)
+        return unique_imports
+
+    def get_missing_test_imports(self):
+        filters = (
+            self._filter_known_packages,
+            self._filter_only_testing_imports,
+            self._filter_own_package,
+            self._filter_requirements,
+            self._filter_test_requirements,
+            self._filter_python_standard_library,
+        )
+        filtered = self._process_pipeline(self.imports_used, filters)
+        unique_imports = self._get_unique_imports(imports_list=filtered)
+        return unique_imports
+
+    def get_unneeded_requirements(self):
+        all_but_test_requirements = self._requirements.copy()
+        for extra in self._extras_requirements:
+            all_but_test_requirements.update(self._extras_requirements[extra])
+        test_requirements = self._get_test_extra()
+        if test_requirements:
+            for dotted_name in test_requirements:
+                all_but_test_requirements.remove(dotted_name)
+        filters = (
+            self._filter_known_packages,
+            self._filter_python_standard_library,
+            self._filter_used_imports,
+        )
+        unneeded = self._process_pipeline(all_but_test_requirements, filters)
+        unique_imports = self._get_unique_imports(imports_list=unneeded)
+        return unique_imports
+
+    def requirements_that_should_be_test_requirements(self):
+        filters = (
+            self._filter_testing_imports,
+            self._filter_python_standard_library,
+            self._filter_own_package,
+        )
+        non_testing_imports = self._process_pipeline(
+            self.imports_used,
+            filters,
+        )
+        requirements_not_used = [
+            x
```

Comment:
x => requirement

Reply:
Done (as you have seen, usually I tend to use ``x`` on list comprehensions only where the value on each iteration can be easily spotted, but still, I agree that putting a name is always better, I'm lazy on list comprehensions 🙂 )